### ISSUE TITLE
[290-p8] Q4 additional hint

### ISSUE THREAD

**Issue by ymzhang-neo** [TheDataMine/the-examples-book#102]
Dear @kevinamstutz , for Question 4 is it possible to include the number of flights in the description or as a hint? This question involves code that is a bit more complicated, and the total number of flights would allow the students to verify their answer.

Also, could you give me a hint about how to use `echo` (listed under "Relevant topics")? I see it used in [the solution](https://github.com/TheDataMine/fall2020-stat29000/blob/master/project08/stat29000project08solutions.Rmd) under `### 5` - but I am not sure how the solution code for Questions 5-7 correspond to the current version of Project 8.

Just for your reference, my code for Question 4 is:
```
cd /class/datamine/data/flights/subset

# get the airports
awk -F, '{if ($4 ~ /state|AZ|FL/) {print}}' ~/new_airports.csv > ~/az_fl_airports.txt
wc -l ~/az_fl_airports.txt # 160 = 159 + 1 (header)

# get the flights
grep -w -F -f <(cut -d, -f1 ~/az_fl_airports.txt) 2008.csv > ~/az_fl_flights_2008.csv
wc -l ~/az_fl_flights_2008.csv # 484705
```  
I thought that we need [process substitution](https://tldp.org/LDP/abs/html/process-sub.html) to first obtain the IATA codes for `grep`, but this is not used in the solution. Could you help me clarify it? Thanks!!

**Comment by kevinamstutz** [comment]
Dear @ymzhang-neo, yes absolutely! I'll add a hint where they can verify their answer. I'm going to remove echo, that was a solution to an old question that I must have removed but forgot to update the solutions before I pushed them to github. I'll update my solutions as well.

Your solution works! My solution was just to do:

```bash
awk -F, '{if ($4 == "AZ" || $4 == "FL") print $1}' new_airports.csv > az_fl_airports.txt
grep -w -F -f az_fl_airports.txt 2008.csv > az_fl_airports.txt
wc -l az_fl_airports.txt
```

I don't think you need to use process substitution here, just regular old redirection should work. I think process substitution was just sort of how `comm` worked.

**Comment by ymzhang-neo** [comment]
Dear @kevinamstutz , thanks for the the update! 

I realize that I misunderstood the question
> `az_fl_airports.txt` should contain a list of airport codes for all airports from both Arizona (AZ) and Florida (FL).

I thought it was okay to keep all the columns, and thus I tried the process substitution. Can we update the description as follows?
> `az_fl_airports.txt` should _only_ contain a list of airport codes for all airports from both Arizona (AZ) and Florida (FL).

**Comment by kevinamstutz** [comment]
Fixed in 8b12a83, great point!
